README.md

```diff
@@ -1 +1,200 @@
-# dp-cache
\ No newline at end of file
+# dp-cache
+
+dp-cache is a library which provides functionality for a service to cache data by creating a cache object which contains the data (stored in memory). The update interval is used to define how often the cache should be updated and the use of Go channels for the application to update the cache object.
+
+```go
+// Cache contains all the information to start, update and close caching data
+type Cache struct {
+ data        sync.Map
+ config      Config
+ close       chan struct{}
+ UpdateFuncs map[string]func() (interface{}, error)
+}
+```
+
+## Setup app to cache data
+
+Please note that the data is stored in a `sync.Map` which is like a Go `map[interface{}]interface{}` but is safe for concurrent use by multiple goroutines without additional locking or coordination [[1]][golang-sync-map].
+
+### 1. Initialising the cache
+
+- Create a `cache` package in the app
+- Within the package, create a file which is appropriately named to the data which is being cached
+
+#### a. Add a wrapper NEW cache function
+
+The library gives us the option to cache any form of data. We want to create a wrapper function which allows us to extend the cache object for our needs.
+
+##### Example
+
+```go
+import (
+    dpcache "github.com/ONSdigital/dp-cache"
+)
+```
+
+```go
+// TopicCache is a wrapper to dpcache.Cache which has additional fields and methods specifically for caching topics
+type TopicCache struct {
+    *dpcache.Cache
+}
+```
+
+```go
+// NewTopicCache create a topic cache object to be used in the service which will update at every updateInterval
+// If updateInterval is nil, this means that the cache will only be updated once at the start of the service
+func NewTopicCache(ctx context.Context, updateInterval *time.Duration) (*TopicCache, error) {
+    config := dpcache.Config{
+        UpdateInterval: updateInterval,
+    }
+
+    cache, err := dpcache.NewCache(ctx, config)
+    if err != nil {
+        logData := log.Data{
+            "update_interval": updateInterval,
+        }
+        log.Error(ctx, "failed to create cache from dpcache", err, logData)
+        return nil, err
+    }
+
+    return &TopicCache{cache}, nil
+
```

Comment:
You have two returns now; I would suggest removing 61 and 62

Reply:
Done